File: leo2-common/src/main/java/org/deku/leo2/rest/services/internal/v1/TestService.java
package org.deku.leo2.rest.services.internal.v1;

import org.deku.leo2.rest.entities.internal.v1.TestEntry;

import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;

/**
 * Test service
 * Created by masc on 23.07.14.
 */
@Path("internal/v1/test")
@Consumes(MediaType.APPLICATION_JSON)
public interface TestService {
    /** Get all tours */
    @GET
    @Path("/")
    TestEntry[] get();

    /** Get entry by name
     * @param name name */
    @GET
    @Path("/{name}")
    TestEntry get(@PathParam("name") String name);

    /** Find entries
     * @param query Query string */
    @GET
    @Path("/find")
    TestEntry[] find(@QueryParam("q") String query);
}

File: leoz-common/src/main/java/org/deku/leoz/rest/service/internal/v1/TestService.kt
package org.deku.leoz.rest.service.internal.v1

import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import io.swagger.annotations.ApiParam
import org.deku.leoz.rest.entity.internal.v1.ApplicationVersion
import javax.ws.rs.*
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

/**
 * Created by masc on 09.10.15.
 */
@Path("internal/v1/test")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
@Api(value = "Test operations")
interface TestService {
    /**
     * Get entry by name
     * @param name name
     */
    @GET
    @Path("/test-soap-call")
    @ApiOperation(value = "Invoke soap call for testing")
    fun testSoapCall(): Response
}